specs/ssh_configuration.adoc

```diff
@@ -0,0 +1,271 @@
+u'key_options':None,
+u'state':u'present',
+u'user':u'tendrl',
+u'key':u'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC5hqcQ0EkbEsnowSdh8B2RPuvgEfIFvyT4
+        /cbHODYc+/y7OxOK66J7006t9llzmhCG6jYE5jCZz4Q+rKHQ9G6+0Fc2Av7x9KdnEeYkBH8a2
+        HgmyA16dJW+eTLL4QK4TLC03yFSedorc9NXbCaYW0EMc59lZMUiXaGUtzAjm+QLfh7h8Vf9Z
+        4fNJkj8zURIbfemHaoYgv6ElJGg+sdKAhZwULRe0ZzrYsRXBSPKQCfzavnt7XXi/VWJTlirh
+        lAp7WR97M26ufEN7bPB27wIcF4FZUo8kLlULRn58VB0rIXXqNKM2BI+v4y9UvCmeqlTFFGuVx
+        0Ep+trrYDcsqwuulyp ansible-generated on dhcp35-60.lab.eng.blr.redhat.com',
+u'invocation':{
+  u'module_args':{
+    u'exclusive':False,
+    u'keyfile':u'/home/tendrl/.ssh/authorized_keys',
+    u'key_options':None,
+    u'state':u'present',
+    u'user':u'tendrl',
+    u'key':u'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC5hqcQ0EkbEsnowSdh8B2RPuvgEfIF
+            vyT4/cbHODYc+/y7OxOK66J7006t9llzmhCG6jYE5jCZz4Q+rKHQ9G6+0Fc2Av7x9KdnE
+            eYkBH8a2HgmyA16dJW+eTLL4QK4TLC03yFSedorc9NXbCaYW0EMc59lZMUiXaGUtzAjm+
+            QLfh7h8Vf9Z4fNJkj8zURIbfemHaoYgv6ElJGg+sdKAhZwULRe0ZzrYsRXBSPKQCfzavn
+            t7XXi/VWJTlirhlAp7WR97M26ufEN7bPB27wIcF4FZUo8kLlULRn58VB0rIXXqNKM2BI+
+            v4y9UvCmeqlTFFGuVx0Ep+trrYDcsqwuulyp ansible-generated on
+            dhcp35-60.lab.eng.blr.redhat.com',
+    u'path':None,
+    u'unique':False,
+    u'validate_certs':True,
+    u'manage_dir':True
+  }
+},
+u'path':None,
+u'unique':False,
+u'keyfile':u'/home/tendrl/.ssh/authorized_keys',
+u'manage_dir':True
+}
+```
+* Create a new utility class called SshUtil in commons with filename ssh_utils.py.
+* Create a method called get_pid to identify the currently running sshd process id.
+```
+ there is 2 ways to find pid:
+  1. "systemctl show sshd.service" grab the value of MainPID.
+     it will give currently running pid but problem is through ansible we can't
+     give pip symbol to filter MainPID using grep.
+  2. Run the command "cat /var/run/sshd.pid" in ansible to get currently running pid
+```
+* Create a function called find_sshd_status in SshUtil class.
+```
+  This method will find port number and ssh daemon status using pid
+  import psutil
+  p = psutil.Process(identified pid from get_pid())
+  p.name() # to print service name nothing but sshd
+  p.connections()
+
+  sample output:
+      [pconn(fd=3, family=10, type=1, laddr=('::1', 22), raddr=('::1', 54960),
+      status='LISTEN')]
+
+      return status and portnumber from this
+```
+* There is one more utility for identifying port is open or not. It will not part of this
+  spec. It will implemented in future.
+
+== Assignee(s)
+
+Primary assignee: Gowtham Shanmugasundaram
+
+== Work Items
+
+https://github.com/Tendrl/specifications/issues/141
+
+== Dependencies
+
+libselinux-python - is a dependency for authorized_key module to copy the key.
+
+== Testing
+
+* Sanity check for flow.
+* Check SSH-key is generated successfully or not
+* Check SSH-key is copied successfully
+* Check the user is created successfully
```

Comment:
May be manual verification of connection to nodes using ssh which doesnt ask passwd.